manifests/init.pp

```diff
@@ -77,6 +78,9 @@
   if $cache_dirs {
     validate_hash($cache_dirs)
   }
+  if $refresh_patterns {
+    validate_hash($refresh_patterns)
```

Comment:
Vox Pupuli are (slowly) getting round to replacing all calls to validate_x with puppet 4 data types.